### PR TITLE
Feature/MS-01/mongoose/postulants

### PR DESCRIPTION
**Pull request by nailaghigo**
**Postulants Entity:**

-Add the following endpoints:

- `GET` => 'app/postulants', => **Get All Postulants**
- `GET` => 'app/postulants/:id', => **Get All Postulants By ID**
- `GET` => 'app/postulants/byLastName/:last_name', => **Get All Postulants By Last Name**
- `POST` => 'app/postulants/add', => **Create a New Postulant**
- `PUT` => 'app/postulants/update/:id', => **Update Postulant Information**
- `DELETE` => 'app/postulants/delete/:id', => **Delete a Postulant**

-Install Mongoose Library and use Queries Like:

- `find()`
- `findById()`
- `findByIdAndUpdate()`
- `findOneAndDelete()`
- `save()`

-Create Postulant Schema

-Use Postman Collection:

postman

### CODE REVIEW

File: controllers/postulants.js [hunk 1]
```diff
@@ -1,205 +1,129 @@
...
-        if (err) { 
-          res.status(500).send(`Error while saving the Postulants with the deleted Postulant`);
-        } else {
-          res.send('Postulant deleted!');
-        }})        
-    } else {
-      res.status(404).send("Error. Check the data sent and try again.");
-    }
-}
+  Postulants.findOneAndDelete({ id: req.params.id })
```
Comment (aquilesmariani) [inline controllers/postulants.js:112]: This removes a Postulant no matter which id you send, try using `findByIdAndDelete`

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Please review the changes, then rebase and we should check that everything is still working as intended

File: README.md [hunk 2]
```diff
@@ -47,11 +47,26 @@ To run this project:
 npm start
 ```
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply README.md:51]: Done!

File: controllers/postulants.js [hunk 3]
```diff
@@ -1,205 +1,129 @@
...
-    last_name: req.query.last_name,
-    gender: req.query.gender,
-    date_of_birth: req.query.date_of_birth,
-    email: req.query.email,
-    phone: req.query.phone,
-    city: req.query.city,
-    state: req.query.state,
-    country: req.query.country,
+  const newPostulant = new Postulants({
+    first_name: req.body.first_name,
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply controllers/postulants.js]: All props names are now updated according to naming conventions mentioned at Slack.

File: controllers/postulants.js [hunk 4]
```diff
@@ -1,201 +1,107 @@
...
+  );
+};
+
+const deletePostulant = (req, res) => {
+  Postulants.findByIdAndDelete(req.params.id)
+    .then(() => {
+      res.send({
+        msg: `Postulant with id: ${req.params.id} was succesfully deleted.`,
+      });
+    })
```
Comment (aquilesmariani) [inline controllers/postulants.js:103]: We should check if the found postulant is null and return a 404
```suggestion
  Postulants.findByIdAndDelete(req.params.id)
    .then((res) => {
      if (!res) return res.status(404).json({ msg: `Postulant with id: ${req.params.id} not found.` });
      return res.send({
        msg: `Postulant with id: ${req.params.id} was succesfully deleted.`,
      });
    })
```

**Review by aquilesmariani — CHANGES_REQUESTED** [review]

File: controllers/postulants.js (hunk 4 above)
Comment (nailaghigo) [reply controllers/postulants.js]: oh ok! thanks for notice that!

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Looks great! Thanks for the changes

**Review by aquilesmariani — APPROVED** [review]

**Pull request merged**